integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs

```diff
@@ -287,3 +287,133 @@ fn cannot_create_pool_from_pool_assets() {
 		);
 	});
 }
+
+#[test]
+fn pay_xcm_fee_with_some_asset_swapped_for_native() {
+	let asset_native = asset_hub_kusama_runtime::xcm_config::KsmLocationV3::get();
+	let asset_one = xcm::v3::Location {
+		parents: 0,
+		interior: [
+			xcm::v3::Junction::PalletInstance(ASSETS_PALLET_ID),
+			xcm::v3::Junction::GeneralIndex(ASSET_ID.into()),
+		]
+		.into(),
+	};
+	let penpal = AssetHubKusama::sovereign_account_id_of(AssetHubKusama::sibling_location_of(
+		PenpalA::para_id(),
+	));
+
+	AssetHubKusama::execute_with(|| {
+		type RuntimeEvent = <AssetHubKusama as Chain>::RuntimeEvent;
+
+		// set up pool with ASSET_ID <> NATIVE pair
+		assert_ok!(<AssetHubKusama as AssetHubKusamaPallet>::Assets::create(
+			<AssetHubKusama as Chain>::RuntimeOrigin::signed(AssetHubKusamaSender::get()),
+			ASSET_ID.into(),
+			AssetHubKusamaSender::get().into(),
+			ASSET_MIN_BALANCE,
+		));
+		assert!(<AssetHubKusama as AssetHubKusamaPallet>::Assets::asset_exists(ASSET_ID));
+
+		assert_ok!(<AssetHubKusama as AssetHubKusamaPallet>::Assets::mint(
+			<AssetHubKusama as Chain>::RuntimeOrigin::signed(AssetHubKusamaSender::get()),
+			ASSET_ID.into(),
+			AssetHubKusamaSender::get().into(),
+			3_000_000_000_000,
+		));
+
+		assert_ok!(<AssetHubKusama as AssetHubKusamaPallet>::AssetConversion::create_pool(
+			<AssetHubKusama as Chain>::RuntimeOrigin::signed(AssetHubKusamaSender::get()),
+			Box::new(asset_native),
+			Box::new(asset_one),
+		));
+
+		assert_expected_events!(
+			AssetHubKusama,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::PoolCreated { .. }) => {},
+			]
+		);
+
+		assert_ok!(<AssetHubKusama as AssetHubKusamaPallet>::AssetConversion::add_liquidity(
+			<AssetHubKusama as Chain>::RuntimeOrigin::signed(AssetHubKusamaSender::get()),
+			Box::new(asset_native),
+			Box::new(asset_one),
+			1_000_000_000_000,
+			2_000_000_000_000,
+			0,
+			0,
+			AssetHubKusamaSender::get()
+		));
+
+		assert_expected_events!(
+			AssetHubKusama,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::LiquidityAdded {lp_token_minted, .. }) => { lp_token_minted: *lp_token_minted == 1414213562273, },
```

Comment:
Do we want to have this hardcoded check for minted lp tokens here?
```suggestion
				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::LiquidityAdded {lp_token_minted, .. }) => { },
```

Reply:
Why not? Not very important in this context, an additional assert is not bad. Here I keep it same as in the existing test above.